optimum/intel/openvino/modeling_open_clip.py

```diff
@@ -0,0 +1,733 @@
+            token=token,
+            local_files_only=local_files_only,
+            force_download=force_download,
+            trust_remote_code=trust_remote_code,
+            ov_config=ov_config,
+            library_name="open_clip",
+            framework="pt",
+            fn_get_submodels=fn_get_submodels,
+            custom_architecture=True,
+            custom_export_configs=custom_export_configs,
+        )
+
+        config.save_pretrained(save_dir_path)
+        return cls._from_pretrained(
+            model_id=save_dir_path,
+            config=config,
+            load_in_8bit=load_in_8bit,
+            quantization_config=quantization_config,
+            file_name=cls.OV_XML_FILE_NAME,
+            **kwargs,
+        )
+
+    @classmethod
+    def _from_pretrained(
+        cls,
+        model_id: Union[str, Path],
+        config: PretrainedConfig,
+        use_auth_token: Optional[Union[bool, str]] = None,
+        token: Optional[Union[bool, str]] = None,
+        revision: Optional[str] = None,
+        force_download: bool = False,
+        cache_dir: str = HUGGINGFACE_HUB_CACHE,
+        file_name: Optional[str] = "openvino_model_text.xml",
+        subfolder: str = "",
+        from_onnx: bool = False,
+        local_files_only: bool = False,
+        load_in_8bit: bool = False,
+        quantization_config: Union[OVWeightQuantizationConfig, Dict] = None,
+        **kwargs,
+    ):
+        return super()._from_pretrained(
+            model_id=model_id,
+            config=config,
+            use_auth_token=use_auth_token,
+            token=token,
+            revision=revision,
+            force_download=force_download,
+            cache_dir=cache_dir,
+            file_name=file_name,
+            subfolder=subfolder,
+            from_onnx=from_onnx,
+            local_files_only=local_files_only,
+            load_in_8bit=load_in_8bit,
+            quantization_config=quantization_config,
+            **kwargs,
+        )
+
+    def forward(
+        self,
+        input_ids: Union[torch.Tensor, np.ndarray],
+        **kwargs,
+    ):
+        self.compile()
+
+        inputs = {"text": input_ids}
+
+        outputs = self._inference(inputs)
+        return ModelOutput(text_features=torch.from_numpy(outputs["text_features"]))
+
+    def _save_pretrained(self, save_directory: Union[str, Path]):
+        dst_path = os.path.join(save_directory, self.OV_XML_FILE_NAME)
+        openvino.save_model(self.model, dst_path, compress_to_fp16=False)
+        generation_config = getattr(self, "generation_config", None)
+        if generation_config is not None:
+            try:
+                generation_config.save_pretrained(save_directory)
+            except Exception as exception:
+                logger.warning(
+                    f"The generation config will not be saved, saving failed with following error:\n{exception}"
+                )
+
+        self._save_openvino_config(save_directory)
```

Comment:
instead of overwriting `_save_pretrained` we add this change here https://github.com/huggingface/optimum-intel/blob/ba74307dfc7f69c6281e811e02ee0203a6fb8355/optimum/intel/openvino/modeling_base.py#L259 instead

Reply:
I'm suggesting to add this change in `OVBaseModel._save_pretrained` directly (with a `default_file_name` attribute that would be `"openvino_model_text.xml"` for `OVModelOpenCLIPText`, `"openvino_model_vision.xml"` for `OVModelOpenCLIPVisual`  and `"openvino_model.xml"` for the rest) wdyt ?

Reply:
I have added `xml_model_name `as attribute to `OVBaseModel` , used it in `_save_pretrained` and overwrited it in `OVModelOpenCLIPText`/`OVModelOpenCLIPVisual` Could you please check if I understand you right ?